pkg/server/heapprofiler/heapprofiler_test.go

```diff
@@ -74,21 +74,25 @@ func testHelper(
 
 func TestPercentSystemMemoryHeuristic(t *testing.T) {
 	rssValues := []rssVal{
-		{0, 30}, {20, 40}, // random small values
-		{30, 88},            // should trigger
-		{80, 89},            // should not trigger as less than 60s before last profile
-		{130, 10}, {140, 4}, // random small values
+		{0, 30}, // random small values
+		{20, 40},
+		{30, 88},  // should trigger
+		{50, 94},  // should not trigger as less than 60s before last profile
```

Comment:
30s, right? Also "after", not "before"

Can you highlight a case in which a trigger occurs because of the 5% threshold?

Reply:
In https://github.com/cockroachdb/cockroach/issues/35985, which I looked into briefly and didn't make any progress on, there's a consistent pattern of a node writing a single heap profile (presumably at 85%) and then crashing 3-4 minutes later, so the memory is increasing faster than 5% every 2 minutes. It might be useful to increase the number of profiles we collect in these cases where the memory is increasing relatively quickly. I've also seen large, sudden increases in allocated memory with the new index backfiller, (I think) related to having to read SSTs received over the network into memory, and I think we've seen some OOMs related to this.

I picked 2 minutes to try to be conservative, but these numbers are pretty arbitrary. We could also just collect a profile every minute/30s and get rid of the 5% increase threshold.

Reply:
Sorry, I meant a test case :-) but this was informational also. I'm less clear on what more heap profiles give you if you already have one at 85% but I assume you have a bunch of memory there but maybe not a single large source of allocation, and you just want to take the guesswork out of it by getting an idea of the rate of allocation of the sources. I thought `alloc_{space,objects}` gave you that in a single profile already, though (by telling you the number of allocations in one GC period).

Reply:
Yeah, that's the intent - I was just writing a follow up comment to that effect, but in the case of e.g. #35985 there's no obviously problematic source of allocation, and it might be useful to know what's large and constant and what's increasing.

I thought `alloc_space` included every allocation since the process started. Is that not the case?

And I'll update the test shortly.